libs/sdk-backend-tiger/src/backend/workspace/facts/index.ts

```diff
@@ -1,14 +1,48 @@
-// (C) 2019-2020 GoodData Corporation
-import { IWorkspaceFactsService, IMetadataObject } from "@gooddata/sdk-backend-spi";
-import { ObjRef, idRef } from "@gooddata/sdk-model";
-import { newDataSetMetadataObject } from "@gooddata/sdk-backend-base";
+// (C) 2019-2021 GoodData Corporation
+import { IDataSetMetadataObject, IMetadataObject, IWorkspaceFactsService } from "@gooddata/sdk-backend-spi";
+import { isIdentifierRef, ObjRef } from "@gooddata/sdk-model";
+import { TigerAuthenticatedCallGuard } from "../../../types";
+import { ITigerClient, jsonApiHeaders } from "@gooddata/api-client-tiger";
+import { invariant } from "ts-invariant";
+import { convertDatasetWithLinks } from "../../../convertors/fromBackend/MetadataConverter";
 
 export class TigerWorkspaceFacts implements IWorkspaceFactsService {
-    constructor() {}
+    constructor(private readonly authCall: TigerAuthenticatedCallGuard, public readonly workspace: string) {}
 
-    public async getFactDatasetMeta(_ref: ObjRef): Promise<IMetadataObject> {
-        return newDataSetMetadataObject(idRef("dummyDataset"), (m) =>
-            m.id("dummyDataset").uri("/dummy/dataset").title("Dummy dataset").description(""),
-        );
+    public async getFactDatasetMeta(ref: ObjRef): Promise<IMetadataObject> {
+        return this.authCall((client) => {
+            return loadFactDataset(client, this.workspace, ref);
+        });
     }
 }
+
+function loadFactDataset(
+    client: ITigerClient,
+    workspace: string,
+    ref: ObjRef,
+): Promise<IDataSetMetadataObject> {
+    invariant(isIdentifierRef(ref));
```

Comment:
We should probably add a message here as this can be triggered by an external user.